File: android/entities/src/main/java/com/tiagohs/entities/tmdb/movie/MovieExtraInfo.kt
package com.tiagohs.entities.tmdb.movie

import com.google.gson.annotations.SerializedName
import com.tiagohs.entities.Quote
import com.tiagohs.entities.contents.ContentBlockSpecial
import com.tiagohs.entities.contents.ContentQuote
import com.tiagohs.entities.main_topics.MainTopic
import com.tiagohs.entities.main_topics.MainTopicItem
import com.tiagohs.entities.main_topics.MilMoviesMainTopic
import java.io.Serializable

data class MovieExtraInfo (
	@SerializedName("id") val id : Int,
	@SerializedName("review_results") val reviewResults : List<ReviewsResult>? = null,
	@SerializedName("quote") val quote : ContentQuote? = null,
	@SerializedName("watchOn") val watchOn : List<Network>? = null,
	@SerializedName("did_you_know_list") val didYouKnowList : List<DidYouKnow>? = null,
	@SerializedName("block_special") val blockSpecial : ContentBlockSpecial? = null,
): Serializable {
	var milMoviesMainTopic: MilMoviesMainTopic? = null
	var historyMainTopic: MainTopicItem? = null
}

File: ios/HistoryCinema/HistoryCinema/Entities/LocalFiles/Movie/MovieExtraInfo.swift
//
//  MovieExtraInfo.swift
//  HistoryCinema
//
//  Created by Tiago Henrique da Silva on 27/02/21.
//

import Foundation
import ObjectMapper

class MovieExtraInfo: BaseLocalModel {
    var id : Int!
    var reviewResults : [ReviewsResult]?
    var quote : ContentQuote?
    var watchOn : [LocalNetwork]?
    var didYouKnowList : [DidYouKnow]?
    var blockSpecial : ContentBlockSpecial?

    var milMoviesMainTopic: MilMoviesMainTopic?
    var historyMainTopic: MainTopicItem?
    
    required init?(map: Map) {
        super.init(map: map)
    }

    override func mapping(map: Map) {
        id                                                      <- map["id"]
        reviewResults                                           <- map["review_results"]
        quote                                                   <- map["title"]
        watchOn                                                 <- map["watchOn"]
        didYouKnowList                                          <- map["did_you_know_list"]
        blockSpecial                                            <- map["block_special"]
    }
}
